core/devmode/src/main/java/io/quarkus/dev/CompilerFlags.java

```diff
@@ -0,0 +1,85 @@
+package io.quarkus.dev;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.wildfly.common.annotation.NotNull;
+import org.wildfly.common.annotation.Nullable;
+
+import io.quarkus.runtime.util.StringUtil;
+
+/**
+ * A set of compiler flags for javac.
+ *
+ * Can combine system-provided default flags with user-supplied flags and <code>-source</code>
+ * and <code>-target</code> settings.
+ */
+public class CompilerFlags {
+
+    @NotNull
```

Comment:
I don't think these annotations provide any benefit. Or am I missing something?

Reply:
They document the invariants of the class.  But they don’t do anything, so if you prefer, I’ll remove them.

Reply:
Done.

Reply:
That's what I assumed, but since we aren't using them elsewhere, best be uniform.